cmd/harborAPI.go

```diff
@@ -435,3 +436,75 @@ func Catalogit(container CatalogitContainer) (response string, err []error) {
 
 	return "Successfully Cataloged Container " + container.Name, nil
 }
+
+//IsContainerVersionCataloged determines whether or not a container/version exists in the catalog
+func IsContainerVersionCataloged(name string, version string) bool {
+
+	//build URI
+	values := make(map[string]interface{})
+	values["name"] = name
+	values["version"] = version
+	template, _ := uritemplates.Parse(customsURI + "/catalog/{name}/{version}/")
+	uri, _ := template.Expand(values)
+	if Verbose {
+		log.Println("fetching: " + uri)
+	}
+
+	//issue request
+	res, _, err := gorequest.New().Get(uri).EndBytes()
+
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	//not found
+	if res.StatusCode == 404 {
+		return false
+	}
+
+	//throw error if not 200
+	if res.StatusCode != 200 {
+		log.Fatalf("GET %v returned %v", uri, res.StatusCode)
+	}
+
+	return true
+}
+
+// Deploy deploys (and catalogs) a shipment container to an environment
+func Deploy(shipment string, env string, buildToken string, deployRequest DeployRequest) {
+
+	//build URI
+	values := make(map[string]interface{})
+	values["shipment"] = shipment
+	values["env"] = env
+	template, _ := uritemplates.Parse(customsURI + "/deploy/{shipment}/{env}/ec2")
```

Comment:
we should not hardcode location here. Needs to be passed in as well with shipment information.